### ISSUE TITLE
Pose - human - 人体姿态估计

### ISSUE THREAD

**Issue by junxnone** [junxnone/tio#291]
# Reference
- [ ] [人体姿态估计综述（Human Pose Estimation Overview）](https://blog.csdn.net/qq_36165459/article/details/78320535)
- [ ] [基于部分亲和字段PAF(Part Affinity Field)的2D图像姿态估计](https://blog.csdn.net/qq_14845119/article/details/72526737)
- [ ] [Action Recognition training with opencv/training_toolbox_caffe](https://github.com/opencv/training_toolbox_caffe#action-recognition-training)
- [ ] [人体姿态识别研究综述 - fengfeng，Z ](https://blog.csdn.net/qq_38522972/article/details/82953477)
- [ ] [人体姿态估计(Human Pose Estimation)文献综述](https://blog.csdn.net/BockSong/article/details/81037059)
- [ ] [Human Pose Estimation人体姿态估计综述调研](https://blog.csdn.net/anymake_ren/article/details/81978260)
- [ ] [行为识别的一些资料](https://github.com/jindongwang/activityrecognition)

# 2D姿态估计方法分类
- Bottom-up，先catch所有关节，再关联人，所需计算时间基本不变
- Top-down，先检测人的bounding box，再用single的方法检测每个人的关节，运算时间会随着图像中人的个数而显著增加。

![image](https://user-images.githubusercontent.com/2216970/52159727-62e33580-26e3-11e9-923a-fa988e2edade.png)

# Framework
Framework | Update | Support | Dependencies
-- | -- | -- | --
VideoPose3D | updating | -- | <ul><li>Python3 +</li><li>PyTorch</li></ul>
DetectAndTrack | updating | -- | Nvidia GPU
OpenPose-Plus | updating |
OpenPose |  updating | Framework<ul><li>TensorFlow</li><li>OpenPose C++</li><li>Caffe2</li><li>PyTorch</li><li>MXNet</li><li>CNTK</li><li>Chainer</li><li>MatConvnet</li></ul> | <ul><li>OpenCV</li><li>Caffe</li></ul>
DensePose |  updating |
hpep #334 |  updating |
RMPE #296 | updating | -- | the caffe for train is too old (2016)
AlphaPose |  updating | -- | <ul><li>General version</li><ul><li>Torch</li><li>Tensorflow</li></ul><li>PyTorch version</li><ul><li>PyTorch</li></ul></ul>
pose-tensorflow | 2017|<ul><li>MPITrain</li><li>COCO</li><li>Your own dataset</li></ul>
DeepPose | 2017

# Train model

# 评估
主要评价标准包括以下两种，PCK的评价标准多出现在单人的姿态估计上，多人的姿态估计评价标准基本上为mAP，比赛和学术论文中mAP成为衡量结果的必须指标。

PCK 
Percentage of Correct Keypoints (PCK)，reports the percentage of keypoint detection falling within a normalized distance of the ground truth. 
For FLIC, distance is normalized by torso size, and for MPII, by a fraction of the head size (referred to as PCKh).

mAP 
人体骨骼关键点的评价指标类比通用的物体检测评价方式，将最终的mAP(mean Average Precision)值作为评价依据。物体检测任务中使用IoU(Intersection over Union)来评价预测与真实标注之间的差异，在人体骨骼关键点检测任务中，我们使用OKS(Object Keypoint Similarity)代替IoU，对选手预测的人体骨骼关键点位置与真实标注之间的相似性进行打分。
![image](https://user-images.githubusercontent.com/2216970/52159710-2283b780-26e3-11e9-8fc9-003df2cf1d1f.png)

# Term
pairwise relations ( PAF - part affinity fields) ：
heatmaps：
